### DISCUSSION TITLE
تعديل طريقة المساهمة

### DISCUSSION THREAD

**Discussion by X7md** [SaudiOpenSourceCommunity/SaudiOSS#62] [Ideas]
# تعديل طريقة المساهمة

طريقة المساهمة وإضافة المشاريع الحالية غير جيدة

https://github.com/SaudiOpenSourceCommunity/SaudiOSS/blob/71ea299b01bedde6564ec7cc4c439038f7286167/homepage/src/githubApi.js#L12-L43

بدلًا من جعل المُساهم يضيف مشاريعه `as HTML` أليس من الأفضل جعله يقوم بوضعها `as JSON` مثلًا؟ 
وعلى المساهم فقط تعديل الملف **بإضافة مصفوفة مشاريعه**

```json
[
    {
        "name": "اسم الشخص",
        "ghUsername": "مُعرف الشخص",
            "projects" : [
                { 
                  "title": "عنوان المشروع",
                  "url": "رابط المشروع",
                  "desc": "وصف عن المشروع"
                },
                { 
                  "title": "عنوان المشروع",
                  "url": "رابط المشروع",
                  "desc": "وصف عن المشروع"
                }
            ]
    }
]
```

ثم من خلال Github Actions حين قُبول الطلب يتم توليد ملف Readme.md مُعدل ويُحدث في المستودع

**Comment by Alaboudi1** [comment]
لسبب ما ما انتبهت للأقتراح هنا. اقتراحك جيد لكن معقد بعض الشي. اذا عندك قدرة تنفذ الفكرة وتضيفها للمشروع نكون لك شاكرين :)

**Comment by mo9a7i** [comment]
تم التنفيذ في #75
